### PR TITLE
Display app version

### PR DESCRIPTION
**Pull request by Gbarra9**
This pull request is related to the following:
- https://github.com/episphere/connect/issues/621
 
Related pull requests:
- https://github.com/episphere/studyManagerDashboard/pull/3
- https://github.com/episphere/connectApp/pull/813

Problem: Display the app version number

Code changes ([from Warren's push and other small fixes](https://github.com/episphere/biospecimen/pull/730/commits/5ef27c3175a0c44d0ab129e0c44a1a47b58d5034)):
- In `index.html`, script for using sentry removed
- In `index.js`, service worker conditional with listeners to check if service worker is registered and to have main thread and service worker to communicate to get the app version
- In `serviceWorkers.js`, add version number be in a constant; add cache version number in register route for `NetworkFirst` and `CacheFirst` Strategy; include listener to install a new service worker immediately skipping waiting phase , listener to remove old cache when new service worker activates, and event listener to respond to the main thread asking for "getAppVersion" to display version number to footer

### CODE REVIEW

**Review by amber-emmes — COMMENTED** [review]
This looks good, but wasn't it also supposed to be used in the service worker file?

**Comment by we-ai** [comment]
I feel that hard-coded version isn't the best approach, because a hard-coded version value may not be correct and each change on version value needs code change. In this particular PR of biospecimen repo, version 1.4.0 refers to May (2024) release, and the July release will have a different version number which hasn't been decided yet.
To minimize errors of version mismatches, one way might be making versions clear and straightforward (either to website users or to us). For example, if July version is 24.07 (or 24.7.0 etc), users will easily know the meaning of version. Or if we want to stick to our current version conventions, we can use intermediate values (eg release date 20240731) to map to final version values.

**Comment by anthonypetersen** [comment]
@we-ai 

I'm open to whatever version numbers we want to use, my idea in general though was that the version number would be updated with every release and the change would be included in the work the senior developer does who is in charge of the release tasks for that month. 

Do you have a suggestion for maintaining a dynamic version number that needs to change every release?

**Comment by we-ai** [comment]
> @we-ai
> Do you have a suggestion for maintaining a dynamic version number that needs to change every release?

The easiest way I can think of is having a version format that's clear and straightforward, so that everybody can understand it and update it if needed.

Based on above criteria, format `24.7` can be a candidate, meaning formal release in July, 2024. If some minor fixed are pushed after July release, the version can be `24.7.1`, `24.7.2` etc.

File: index.js [hunk 1]
```diff
@@ -136,3 +137,13 @@ const userLoggedIn = () => {
...
+
+(function displayAppVersion() {
+    document.addEventListener('DOMContentLoaded', function() {
+        const contentWrapper = document.querySelector('.footer-content .content-wrapper');
+
+        const newElement = document.createElement('div');
+        newElement.innerHTML = `<p id="appVersion">${appVersion.versionNumber}</p>`;
+        contentWrapper.appendChild(newElement);
+    });
+})();
```
Comment (we-ai) [inline index.js:149]:
```suggestion
    document.addEventListener('DOMContentLoaded', function() {
        const contentWrapper = document.querySelector('.footer-content .content-wrapper');

        const newElement = document.createElement('div');
        newElement.innerHTML = `<p id="appVersion">${appVersion.versionNumber}</p>`;
        contentWrapper.appendChild(newElement);
    });
```

File: appVersion.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const appVersion = {
+    "versionNumber": "v1.4.0"
```
Comment (we-ai) [inline appVersion.js:2]:
```suggestion
    "versionNumber": "v24.7.0"
```

File: index.js [hunk 3]
```diff
@@ -49,15 +66,7 @@ const datadogConfig = {
...
-window.onload = () => {
-    if ("serviceWorker" in navigator) {
-        try {
-            navigator.serviceWorker.register("./serviceWorker.js");
-        } catch (error) {
-            console.log(error);
-        }
-    }
-
+window.onload = async () => {
```
Comment (anthonypetersen) [inline index.js:69]: is this needed?

**Review by amber-emmes — APPROVED** [review]
Looks good!

**Review by anthonypetersen — APPROVED** [review]

**Review by we-ai — APPROVED** [review]

**Pull request merged**